internal/service/federatedsettingsorgconfig/resource_federated_settings_connected_org.go

```diff
@@ -60,7 +60,7 @@ func Resource() *schema.Resource {
 
 func resourceMongoDBAtlasFederatedSettingsOrganizationConfigRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	// Get client connection.
-	conn := meta.(*config.MongoDBClient).Atlas
+	conn := meta.(*config.MongoDBClient).AtlasV2
 
 	if d.Id() == "" {
 		d.SetId("")
```

Comment:
short answer: Not sure why, but it's beyond the scope of the PR which would be better to keep them focused to single topic.

long answer: might be trivial to remove but I am not sure if there's any weird difference behind it. Looking at the code:
```
// SetId sets the ID of the resource. If the value is blank, then the
// resource is destroyed.
func (d *ResourceData) SetId(v string) {
	d.once.Do(d.init)
	d.newState.ID = v

	// once we transition away from the legacy state types, "id" will no longer
	// be a special field, and will become a normal attribute.
	// set the attribute normally
	d.setWriter.unsafeWriteField("id", v)

	// Make sure the newState is also set, otherwise the old value
	// may get precedence.
	if d.newState.Attributes == nil {
		d.newState.Attributes = map[string]string{}
	}
	d.newState.Attributes["id"] = v
}


// Id returns the ID of the resource.
func (d *ResourceData) Id() string {
	var result string

	if d.state != nil {
		result = d.state.ID
		if result == "" {
			result = d.state.Attributes["id"]
		}
	}

	if d.newState != nil {
		result = d.newState.ID
		if result == "" {
			result = d.newState.Attributes["id"]
		}
	}

	return result
}
```

Without solid testing, it's an unnecessary risk the one of removing this statement (and, as I said, not related to the title of this PR).

Reply:
There are two similar cases in the code base where there is an `if d.Id() == "" {d.setId("")}` - we should figure why this is done and make sure there are tests around it before changing it. @lantoli do you want to create a Jira issue for that?

Reply:
actually let me just create that Jira issue, thank you for calling that out.

Reply:
thanks